course_catalog/filters.py

```diff
@@ -39,3 +42,30 @@ def filter_audience(self, queryset, _, value):
         else:
             queryset = queryset.exclude(platform__in=PROFESSIONAL_COURSE_PLATFORMS)
         return queryset
+
+    def filter_withcertificate(self, queryset, _, value):
+        """Certificate filter for courses"""
+        if value == "":
+            return queryset
+        else:
+            qs1 = queryset.filter(
+                runs__availability__in=[
+                    AvailabilityType.current.value,
+                    AvailabilityType.upcoming.value,
+                    AvailabilityType.starting_soon.value,
+                ],
+                platform=PlatformType.mitx.value,
+            )
+            qs2 = queryset.filter(
```

Comment:
This logic longer matches the existing logic for determining the certificate property here
https://github.com/mitodl/open-discussions/blob/master/course_catalog/models.py#L320. We should either update that logic to match or change the query here. I think the original makes sense since archived mitx courses are available to audit but professional courses are not. Did you discuss this change with Peter?

Reply:
That wasn't the intent, I think I just muddled logic trying to make it match that logic which is - if it's in one or the other, and not archived anywhere - I believe i updated it correctly now